core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -860,6 +864,16 @@ private void updateGraphHooks(final GraphConfig config) {
                     config.getHooks().add(0, new NamedOperationResolver());
                 }
             }
+            boolean hasNamedViewHook = false;
+            for (final GraphHook graphHook : config.getHooks()) {
+                if (NamedViewResolver.class.isAssignableFrom(graphHook.getClass())) {
+                    hasNamedViewHook = true;
+                    break;
+                }
+            }
+            if (!hasNamedViewHook) {
+                config.getHooks().add(0, new NamedViewResolver());
```

Comment:
I think we need the NamedViewResolver to be after the NamedOperationResolver. After named operations have been resolved, we then need to resolve any named views that exist within the named operations.